---
title: "Pipeline Integrated R Analysis Tool (PIpeRAT) for 16S NGS Bioinformatics Data Analysis" 
output: 
  html_document:
    toc: true
    theme: cerulean
---
**About PIpeRAT:**  
PIpeRAT is a tool for generating the Operational Taxinomic Unit (OTU) table used for downstream analysis of 16S Next Generation Sequince (NGS) bioinformatics data.  This tool integrates a system of preprocessing and quality control steps in R and produces a BIOM table that can then be used to analyze your 16S NGS data.  Several open-source scripts are used PIpeRAT, in addition to R packages.  These will need to be downloaded, compiled, and added to your $PATH prior to running PIpeRAT.  Detailed instructions for setting up the dependencies necessary to run PIpeRAT are included in the readme file.

#System and R version Information
This document shows you the location of PIpeRAT system dependencies after you have specified the correct locations based on your individual machine.  Running the pipeline.Rmd script also sources all of the PIpeRAT scripts and produces the .biom OTU table as well as some inital Alpha and Beta analyses from MacQIIME.    

```{r, echo=TRUE,eval=TRUE,include=TRUE,messages=TRUE,error=TRUE}
ptm <- proc.time()
```

##System Information
```{r,echo=F}
Sys.info()
```

##R Version
```{r,echo=F}
R.Version()
```

```{r include=F}
library("rPython")
library("ShortRead")
library("Biostrings")
library("data.table")
```

#MacQIIME and Sickle Script Directories
##MacQIIME
```{r,echo=F}
Sys.which("macqiime")
```
##Sickle
```{r,echo=F}
Sys.which("sickle")
```

#Machine Dependencies
```{r}
if(Sys.getenv("LOGNAME")=="puruckertom"){ #macqiime is on root on the mac hard drive
  pipeline.dir<-path.expand("~/Dropbox/pipeline/")
  fastq.dir <- "/usr/local/bin/"
  fastq_join.dir <- "/macqiime/bin/fastq-join"
  qiime.dir <- "/macqiime/bin/" 
  py_join <- "/macqiime/bin/"
}
if(Sys.info()[4]=="kens-air"){
  pipeline.dir<-path.expand("~/git/pipeline/")
  sickle.dir <- "/usr/bin/"
  fastq_join.dir <- "/macqiime/bin/fastq-join"
  qiime.dir <- "/macqiime/bin/"
  py_join <- "/macqiime/bin/"
  qpy <- "/macqiime/bin/python "
  seq_filter.dir <- "/Users/KenBradshaw/Desktop/sfbr_metagenomic_example/sfbr_fastq_samples/RDPTools/SeqFilters/dist/"
  headspace_remove.dir <- "/Users/KenBradshaw/git/pipeline"
  usearch.dir <- "/Users/KenBradshaw/git/pipeline/"
  map.dir <- "/Users/KenBradshaw/git/pipeline/SFBR_mapping_files/"
}
if(Sys.info()[4]=="Kens-MacBook-Air.local"){
  pipeline.dir<-path.expand("~/git/pipeline/")
  sickle.dir <- "/usr/bin/"
  fastq_join.dir <- "/macqiime/bin/fastq-join"
  qiime.dir <- "/macqiime/bin/"
  py_join <- "/macqiime/bin/"
  qpy <- "/macqiime/bin/python "
  seq_filter.dir <- "/Users/KenBradshaw/Desktop/sfbr_metagenomic_example/sfbr_fastq_samples/RDPTools/SeqFilters/dist/"
  headspace_remove.dir <- "/Users/KenBradshaw/git/pipeline"
  usearch.dir <- "/Users/KenBradshaw/git/pipeline/"
  map.dir <- "/Users/KenBradshaw/git/pipeline/SFBR_mapping_files/"
}
```

```{r include=F}
file.exists(pipeline.dir)
file.exists(sickle.dir)
file.exists(fastq_join.dir)
file.exists(qiime.dir)
```


#Working Directory
```{r,echo=TRUE}
setwd(pipeline.dir) 
getwd()
```
#Data and Script Directories
```{r include=F}
fastq.unfiltered.dir <- paste(pipeline.dir,"SFBR_Data/",sep="") #Raw NGS Data Directory
fastq.filtered.dir <- paste(pipeline.dir,"SFBR_Data_filtered/",sep="") #Filtered Data Directory
filter.countseqs.dir <- paste(pipeline.dir,"Filter_Seq_Counts/",sep="")#Directory to store pre and post trimmed seq counts
unfiltered.seqcounts.dir<- paste(filter.countseqs.dir,"unfiltered_seq_counts/",sep="") #Unfiltered seq counts
filtered.seqcounts.dir<- paste(filter.countseqs.dir,"filtered_seq_counts/",sep="") #Filtered seq counts
trimmed.singles.dir <- paste(fastq.filtered.dir,"trimmed.singles/",sep="")
fastq.paired.dir <- paste(pipeline.dir,"SFBR_Data_paired/",sep="") #Paired Ends Data Directory
fastq.paired.write <- paste(fastq.paired.dir,"SFBR_joined",sep="") #Directory to write joined file names to
fastq.scripts.dir <- paste(pipeline.dir,"SFBR_Scripts/",sep="") #Directory with join paired ends scripts for each sample
fastq.convert.fasta.dir <- paste(pipeline.dir,"SFBR_paired_fastq_to_fasta/",sep="") #Directory with qual and fna names for each sample
fastq.paired.join.dir <- paste(pipeline.dir,"SFBR_joined",sep="") #Directory with joined paired ends file names
filtered.dir <- paste(pipeline.dir,"SFBR_SeqFiltered/",sep="") #Directory to save SeqFiltered files
SFBR_notag.write.dir <- paste(pipeline.dir,"SFBR_notag",sep = "") #Directory to save removed headspace files to
modified.write.dir <- paste(pipeline.dir,"SFBR_modified_headspace/",sep = "") #Directory to copy removed headspace files to
rmchimera.dir <- paste(pipeline.dir,"SFBR_no_chimeras/",sep ="") #Directory to save removed chimeras files to
subsamp.dir<-paste(pipeline.dir,"SFBR_no_chimeras_subsampled/",sep="") #Directory to store random subsampled reads
subsamp_files_seqs.dir<-paste(subsamp.dir,"SFBR_subsampled_wseqs/",sep="") #Directory to store unique subsampled sequence reads
seqs_in.dir<-paste(pipeline.dir,"SFBR_seqs_in_otu/",sep="") #Directory to store denovo pick input file (combined sequences)
otu.dir <- paste(pipeline.dir,"otus",sep="") #Directory to save pick denovo information
picked_otu.dir<-paste(otu.dir,"/uclust_picked_otus/",sep="") #Directory to save picked otus
taxa_summary.dir<-paste(otu.dir,"/taxa_summary",sep="") #Directory to save taxa summary information
taxa_summary_plots.dir<-paste(taxa_summary.dir,"/taxa_summary_plots/",sep="") #Directory to save taxa summary plots
taxa_summary_out.dir<-paste(taxa_summary.dir,"/otu/",sep="") #Directoy to save taxa summary outut
arare.dir<-paste(otu.dir,"/alpha_rare/",sep="") #Directroy for alpha analysis
beta_out_dir <-paste(otu.dir,"/beta_dir/",sep="") #Directory for beta analysis
```
```{r}
fastq.unfiltered.dir #Raw NGS Data Directory
fastq.filtered.dir  #Filtered Data Directory
trimmed.singles.dir #Trimmed singles directory
fastq.paired.dir  #Paired Ends Data Directory
fastq.paired.write  #Directory to write joined file names to
fastq.scripts.dir #Directory with join paired ends scripts for each sample
fastq.convert.fasta.dir #Directory with qual and fna names for each sample
fastq.paired.join.dir #Directory with joined paired ends file names
filtered.dir  #Directory to save SeqFiltered files
SFBR_notag.write.dir #Directory to save removed headspace files to
modified.write.dir #Directory to copy removed headspace files to
rmchimera.dir #Directory to save removed chimeras files to
subsamp.dir  #Directory to store random subsampled reads
subsamp_files_seqs.dir #Directory to store unique subsampled sequence reads
seqs_in.dir  #Directory to store denovo pick input file (combined sequences)
otu.dir #Directory to save pick denovo information
picked_otu.dir #Directory to save picked otus
taxa_summary.dir #Directory to save taxa summary information
taxa_summary_plots.dir #Directory to save taxa summary plots
taxa_summary_out.dir #Directoy to save taxa summary outut
arare.dir #Directroy for alpha analysis output
beta_out_dir  #Directory for beta analysis output
```

##Source Pipeline Scripts
```{r 01pipeline, child='01pipeline_filter_low_quality_reads_sickle.Rmd',echo=TRUE,eval=TRUE,include=TRUE,messages=TRUE,error=TRUE}
```
```{r 02pipeline, child='02pipeline_join_paired_ends.Rmd',echo=TRUE,eval=TRUE,include=TRUE,messages=TRUE,error=TRUE}
```
```{r 03pipeline, child='03pipeline_filtersequence.Rmd',echo=TRUE,eval=TRUE,include=TRUE,messages=TRUE,error=TRUE}
```
```{r 04pipeline, child='04pipeline_remove_headspace.Rmd',echo=TRUE,eval=TRUE,include=TRUE,messages=TRUE,error=TRUE}
```
```{r 05pipeline, child='05pipeline_remove_chimera_reads.Rmd',echo=TRUE,eval=TRUE,include=TRUE,messages=TRUE,error=TRUE}
```
```{r 06pipeline, child='06pipeline_Subsample_Sequences.Rmd',echo=TRUE,eval=TRUE,include=TRUE,messages=TRUE,error=TRUE}
```
```{r 07pipeline, child='07pipeline_build_OTU_table.Rmd',echo=TRUE,eval=TRUE,include=TRUE,messages=TRUE,error=TRUE}
```
```{r 08pipeline, child='08pipeline_summarize_taxonomy.Rmd',echo=TRUE,eval=TRUE,include=TRUE,messages=TRUE,error=TRUE}
```
```{r 09pipeline, child='09pipeline_alpha_beta_analysis.Rmd',echo=TRUE,eval=TRUE,include=TRUE,messages=TRUE,error=TRUE}
```
#Run Time
```{r, echo=TRUE,eval=TRUE,include=TRUE,messages=TRUE,error=TRUE}
proc.time() - ptm
```

<!source("01pipeline_filter_low_quality_reads_sickle.Rmd")
source("02pipeline_join_paired_ends.R")
source("03pipeline_filtersequence.R")
source("04pipeline_remove_headspace.R")
source("05pipeline_remove_chimera_reads.R")
source("06Subsample_Sequences.R")
source("07pipeline_build_OTU_table.R")
source("08pipeline_summarize_taxonomy.R")
source("09pipeline_alpha_analysis.R")
>











